java/trace/src/test/scala/org/typelevel/otel4s/java/trace/TracerSuite.scala

```diff
@@ -163,6 +169,26 @@ class TracerSuite extends CatsEffectSuite {
     }
   }
 
+  // typelevel/otel4s#277
+  test("retain all of a provided context through propagation") {
+    TestControl.executeEmbed {
+      for {
+        sdk <- makeSdk(additionalPropagators =
+          Seq(PassThroughPropagator.create("foo", "bar"))
+        )
+        tracer <- sdk.provider.get("tracer")
+        _ <- tracer.joinOrRoot(Map("foo" -> "1", "baz" -> "2")) {
+          for {
+            carrier <- tracer.propagate(Map.empty[String, String])
+          } yield {
+            assertEquals(carrier.size, 1)
+            assertEquals(carrier.get("foo"), Some("1"))
+          }
```

Comment:
maybe create a span too, and propagate inside the span instead? not sure if that adds anything to the test